### ISSUE TITLE
1. Working Group Meeting

### ISSUE THREAD

**Issue by ghost** [Simsso/NIPS-2018-Adversarial-Vision-Challenge#1]
# 1. Working Group Meeting (9. June 2018)

Attendees: Timo Denk, Samed Güner

### Tracks
Research focus on **defense track**. However, state of the art knowledge about attacks is requried in order to validate new defenses. We need to do research in both topics and have to come up with something new at the defense track.

* Defense: Find a function _X->W_ where _X_ is the set of all images and _W_ are the classes. The function is parametrized by _theta_.
* Attack: Find a function _(x1,w,theta)->x2_, which takes an image _x1_ of class _w_ and finds another image _x2_ using the weights _theta_ of the given model. Such that _|x2-x1|_ is minimal and _x2_ is misclassified.

### Evaluation Criterion

1. _Let M be the model and S be the set of samples._
1. _We apply the five best untargeted attacks on M for each sample in S._ Sample from training or test data? Do targeted attacks come into play as well?
1. _For each sample we record the minimum adversarial L2 distance (MAD) across the attacks._ L2 can behave in a weird way (curse of dimensionality). Our test should also be validated using L2 distance.
1. _If a model misclassifies a sample then the minimum adversarial distance is registered as zero for this sample._
1. _The final model score is the median MAD across all samples._
1. _The higher the score, the better._

Our deployment pipeline should perform validation in a very similar manner. In particular L2 distance and median.

Example:  
```
let list of distances d = []
foreach image I_s[i] in the dataset S
    calculate 5 perturbation I_p[j] images from I_s[i]
    foreach image I_p[j]
        calculate |I_p[j]-I_s[i]|_2
    Add minimum L_2 distance to d
return median(d)
```

### Deadlines
* June 25th, 2018: Challenge begins 
* November 1st: Final submission date 
* November 15th: Winners Announced

## Research
We need to do research on both topics. Relevant papers need to be determined, asap.

### Papers

* We must be very strong against [One pixel attack for fooling deep neural networks](https://arxiv.org/pdf/1710.08864.pdf) since its attacks are extremely close in L2.
* [Distillation as a Defense to Adversarial Perturbations against Deep Neural Networks](https://arxiv.org/pdf/1511.04508.pdf)
* [CNNs basic knowledge](http://www.deeplearningbook.org/contents/convnets.html)
* Which activation function is most robust (for instance leaky ReLU parametrization)?

### Ideas
**Linear combinations** of inputs (for evaluation). Determine the distance from an image (when linearly approaching an image of another class) of the first miss-classified input. Analyze how noisy the classifications along the line are.

**Derivative penalties** regularize training with penalities for high first and second order derivatives wrt input changes. 

**Growing filters** of the CNN, similiar to [progressive growing of GANs](https://arxiv.org/pdf/1710.10196.pdf). I have not seen any research that goes into this direction, but it might work. New filters would be _faded in slowly_.

**Fisher information matrix** for network size reduction [Overcoming catastrophic forgetting in neural networks](https://arxiv.org/pdf/1612.00796.pdf). The matrix contains information on how relevant certain weights are for classification.

**Dropout** on kernel level / **additive noise** to kernels of higher layers. This might be common already, we have to do research on that.

## Deployment and Infrastructure
Deployment on AWS or GCP (Azure is _not an option, and never was)_.  
Funding through free tier budget, our own money, and subsequently and in the long run sponsoring by SAP Machine Learning Foundation. Also, there might be sponsored computing power available.

## Miscellaneous
* [GitHub repository](https://github.com/Simsso/NIPS-2018-Adversarial-Vision-Challenge) for code
* [GitHub wiki](https://github.com/Simsso/NIPS-2018-Adversarial-Vision-Challenge/wiki) for documentation
* Watch the [challenge Gitter channel](https://gitter.im/crowdAI/nips-2018-adversarial-vision-challenge)
* Watch the project page to notice once they put more detailed information up
